users/delete_handler.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package users
+
+import (
+	"context"
+	"log/slog"
+	"time"
+
+	"github.com/absmach/magistrala"
+	"github.com/absmach/magistrala/auth"
+	mgclients "github.com/absmach/magistrala/pkg/clients"
+	svcerr "github.com/absmach/magistrala/pkg/errors/service"
+	"github.com/absmach/magistrala/users/postgres"
+)
+
+const defLimit = uint64(100)
+
+type handler struct {
+	clients       postgres.Repository
+	auth          magistrala.AuthServiceClient
+	checkInterval time.Duration
+	deleteAfter   time.Duration
+	logger        *slog.Logger
+}
+
+func NewDeleteHandler(ctx context.Context, clients postgres.Repository, auth magistrala.AuthServiceClient, defCheckInterval, deleteAfter time.Duration, logger *slog.Logger) {
+	handler := &handler{
+		clients:       clients,
+		auth:          auth,
+		checkInterval: defCheckInterval,
+		deleteAfter:   deleteAfter,
+		logger:        logger,
+	}
+
+	go func() {
+		ticker := time.NewTicker(handler.checkInterval)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+				handler.routine(ctx)
+			}
+		}
+	}()
+}
+
+func (h *handler) routine(ctx context.Context) {
+	pm := mgclients.Page{Limit: defLimit, Offset: 0, Status: mgclients.DeletedStatus}
+
+	dbUsers, err := h.clients.RetrieveAll(ctx, pm)
+	if err != nil {
+		h.logger.Error("failed to retrieve users", slog.Any("error", err))
+		return
+	}
+
+	if dbUsers.Total > defLimit {
```

Comment:
Let's fetch and remove in batches, rather than fetching all first then removing, just to avoid edge cases where we slice of removed users are huge.